src/libsyntax_ext/format.rs

```diff
@@ -543,6 +543,10 @@ impl<'a, 'b> Context<'a, 'b> {
         let mut pats = Vec::new();
         let mut heads = Vec::new();
 
+        let names_pos: Vec<_> = (0..self.args.len()).map(|i| {
```

Comment:
This had to go all the way up here because... something to do with E0382. Self gets moved into the expression below with `pieces`? This can go further down if we have a mutable array in the loop and push names onto it, but I assumed we would prefer to avoid mutability.